### PR TITLE
Emit Transfer event in the constructor

### PR DESCRIPTION
**Pull request by deuszx**
I should have bumped minor version but there's already 0.3.0 in `main` that migrated the contracts to ink!5.

### CODE REVIEW

File: data.rs [hunk 1]
```diff
@@ -45,14 +45,23 @@ pub struct PSP22Data {
...
-    pub fn new(supply: u128, creator: AccountId) -> PSP22Data {
+    pub fn new(supply: u128, creator: AccountId) -> (PSP22Data, Vec<PSP22Event>) {
         let mut data = PSP22Data {
             total_supply: supply,
             balances: Default::default(),
             allowances: Default::default(),
         };
         data.balances.insert(creator, &supply);
-        data
+        let events = if supply > 0 {
```
Comment (DamianStraszak) [inline data.rs:55]: There is also another issue here that if `supply` is `0` the balance should not be inserted.

I think the cleanes here is in line `54` just do:
`let events = data.mint(creator, supply)?;`
and get rid of the rest.

Comment (DamianStraszak) [reply data.rs:55]: Also `total_supply` should be initialized at `0` in that case.

Comment (DamianStraszak) [reply data.rs:55]: Now I see that `data.mint(...)` needs an `unwrap()` but that's fine, since it's guaranteed to succeed.

**Comment by deuszx** [comment]
Superseded by #18

**Pull request closed without merging**